extensions/mentions/migrations/2022_10_21_000000_create_post_mentions_group_table.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Schema\Builder;
+
+return [
+    'up' => function (Builder $schema) {
+        $schema->create('post_mentions_group', function (Blueprint $table) {
+            $table->integer('post_id')->unsigned();
+            $table->integer('mentions_group_id')->unsigned();
+            $table->timestamp('created_at')->nullable();
+            $table->primary(['post_id', 'mentions_group_id']);
+
+            $table->foreign('post_id')->references('id')->on('posts')->onDelete('cascade');
+            $table->foreign('mentions_group_id')->references('id')->on('groups')->onDelete('cascade');
+        });
+
+        // do this manually because dbal doesn't recognize timestamp columns
+        $connection = $schema->getConnection();
+        $prefix = $connection->getTablePrefix();
+        $connection->statement("ALTER TABLE `${prefix}post_mentions_group` MODIFY created_at TIMESTAMP NULL DEFAULT CURRENT_TIMESTAMP");
```

Comment:
Can we add a comment that explains why `TIMESTAMP` is needed instead of `DATETIME` ? I don't see anything in the code.

Reply:
I agree that this raw sql should not be part of the migration. Also, dbal is able to recognize timestamp columns just fine, I'm curious what issue you ran into that this was deemed necessary (setting a default)?

Reply:
> I agree that this raw sql should not be part of the migration. Also, dbal is able to recognize timestamp columns just fine, I'm curious what issue you ran into that this was deemed necessary (setting a default)?

See https://github.com/flarum/framework/pull/3658#discussion_r1003169130

Reply:
My concern is mostly about the use of TIMESTAMP and not the raw SQL query. At the latest that code will need to be changed in [15 years because of the TIMESTAMP limit](https://en.wikipedia.org/wiki/Year_2038_problem) so I think an explanation of why a DATETIME couldn't be used would be useful to the person that refactors it in the future, without having to dig for the PR that originally introduced it. And even after reading the comments I'm still not sure what disqualifies DATETIME.

Reply:
Thank you for being persistent with this @clarkwinkelmann - turns out I was just being lazy and following previous migrations and using `timestamp` for no other reason other than "that's how we've done it before" 🙈 

I've now switched to using `dateTime`, which also does not require the raw SQL as well, so win-win :)